File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<int> sec;

        if(s.length() == 0)
        {
            return 0;
        }

        if(s.length() == 1)
        {
            return 1;
        }

        int maxo = 0;

        for(int i = 0; i < s.length(); i++)
        {
            unordered_set<char> hashset;
            hashset.insert(s[i]);
            // int count = 1;
            for(int j = i + 1; j < s.length(); j++)
            {
                if(hashset.find(s[j]) != hashset.end())
                {
                    break;
                }
                
                hashset.insert(s[j]);
                // count++;
            }

            // sec.insert(hashset.size());
            int n = hashset.size();
            maxo = max(maxo,n);
        }
        // auto itr = sec.end();
        // itr--;
        // return (*itr);
        return maxo;
        
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int maxlen = 0;
        int n = s.length();

        for(int i = 0; i < n; i++)
        {
            // HashMap<Character, Integer> map = new HashMap<Character, Integer>();
            HashSet<Character> set = new HashSet<>();

            for(int j = i;  j < n; j++)
            {
                if(set.contains(s.charAt(j)) )
                {
                    break;
                }
                else
                {
                    set.add(s.charAt(j));
                    maxlen = Math.max(maxlen,set.size());
                }
            }

        }
            return maxlen;
        
    }
}